v3/src/components/graph/adornments/count/count-adornment-component.tsx

```diff
@@ -32,6 +38,8 @@ export const CountAdornment = observer(function CountAdornment({ model, cellKey,
   const defaultFontSize = graphModel.adornmentsStore.defaultFontSize
   let fontSize = defaultFontSize
   const prevCellWidth = useRef(plotWidth)
+  const subPlotRegionBoundaries = useRef(adornmentsStore?.subPlotRegionBoundaries(instanceKey, scale) ?? [])
```

Comment:
@kswenson Are the changes in the latest commit what you had in mind? It works except that if I import a document that has a graph with the Count adornment activated, add some movable values to that graph to introduce multiple count instances, then re-import the same document, the multiple count instances remain even though the movable values don't exist. It seems like calling `subPlotRegionBoundaries()` on every render is the only way to fix that, though I could be missing something. And I suppose situations like the one I described may not be common enough to worry about?

Reply:
>It works except that if I import a document that has a graph with the Count adornment activated, add some movable values to that graph to introduce multiple count instances, then re-import the same document, the multiple count instances remain even though the movable values don't exist.

This is the re-hydration problem, which is a problem which we should be worrying about, although we haven't always done so. I looked at the code and it seemed like the `mstReaction` should be sufficient to trigger on re-hydration, but when I looked closer it turned out that the reaction wasn't triggering because its `useEffect` was re-running every time the reaction would have triggered, which meant that the first function (the accessor function) was being called regularly but the reaction function was never being called. This could be fixed by adding `fireImmediately: true`, but I also realized that there was both an `mstReaction` and an `mstAutorun` that were both calling `plotCaseCounts` and that both were generally being reinstalled in these cases which meant that there were redundant calls to `plotCaseCounts`. In the end, I combined them into one `mstAutorun`.

Further testing revealed that there were still times that the `useEffect` which installed the autorun would re-run in situations in which it didn't seem like it needed to, which led to the realization that the `cellKey` is recomputed from scratch on every render, which means it's not stable across renders. Therefore, I memoized the `cellKey` so that only meaningful changes would trigger the effects.

Reply:
@kswenson Thanks 👍 This is all working nicely now. Seems like memoizing `cellKey` should be done for all the other adornment components, right? If so, I'll make a separate chore story for that.

Reply:
>Seems like memoizing cellKey should be done for all the other adornment components, right?

Yes, unless we can figure out a way to handle it at the parent level so it doesn't need to be done in each adornment, but I couldn't think of a simple way to do that so I went with local memoization in this case. Worth thinking a little bit about whether there's a way for the parent to pass down a stable `cellKey` in each case.